stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/common/composer/MessageComposerController.kt

```diff
@@ -123,6 +120,33 @@ public class MessageComposerController(
     private val parentMessageId: String?
         get() = (messageMode as? MessageMode.MessageThread)?.parentMessage?.id
 
+    /**
+     * Gets the current text input in the message composer..
+     */
+    private val messageText: String
+        get() = input.value
+
+    /**
+     * Sets up the data loading operations such as observing the maximum allowed message length.
+     */
+    init {
+        scope.launch {
+            val result = chatDomain.watchChannel(channelId, 0).await()
+
+            if (result.isSuccess) {
+                val channelController = result.data()
+
+                channelController.channelConfig.onEach {
+                    maxMessageLength = it.maxMessageLength
+                }.launchIn(scope)
+
+                channelController.members.onEach { members ->
+                    users = members.map { it.user }
+                }.launchIn(scope)
+            }
```

Comment:
Not sure to do this better in a non-blocking way (without `collect {...}`)

Reply:
I think this is okay. Maybe we can see if these values emit frequently, and if they do, we can debounce them or something, but other than that it seems fine!

Reply:
@MarinTolic any comments on the way we observe the data here is flows?

Reply:
Yes, that's our main problem - how to minimize the number of emits and at the same time provide very generic ViewModels.